e2e/collector_test.go

```diff
@@ -278,7 +252,29 @@ google_cloud:
 				return false, fmt.Errorf("getting collector ConfigMap failed: %w", err)
 			}
 
-			if diff := cmp.Diff(want, cm.Data); diff != "" {
+			got := cm.Data
+
+			// It's impractical to try test complex generated scrape configuration here,
+			// so trim scrapeConfigs section to test external labels and forked google cloud section
+			// config propagation, unless asked otherwise.
+			if trimScrapeConfigs {
```

Comment:
What if we just always remove any `scrape_configs` and remove the `trimScrapeConfigs` variable altogether?